lib/catalog.ex

```diff
@@ -54,6 +56,9 @@ defmodule Catalog do
   def handle_continue({:load_documents, opts}, %__MODULE__{ets: ets} = state) do
     documents_fn = Keyword.get(opts, :documents_fn, &all_documents/0)
 
+    start_time = Time.utc_now()
```

Comment:
You probably want `System.monotonic_time(:millisecond)` instead. This avoids any possible issue where the time on the computer changes while this is measurement is happening.

The other option is to use `:timer.tc` which takes a function and return `{time, return_value}`.

Reply:
Oh great, thanks. Refactoring out a function to contain the loop and using `:timer.tc` seemed nice and clean so I went that route.